mlflow/genai/judges/adapters/databricks_managed_judge_adapter.py

```diff
@@ -87,90 +114,275 @@ def _call_chat_completions(user_prompt: str, system_prompt: str, use_case: str |
             if "use_case" in get_chat_completions_sig.parameters:
                 kwargs["use_case"] = use_case
 
-        return managed_rag_client.get_chat_completions_result(**kwargs)
+        try:
+            return managed_rag_client.get_chat_completions_result(**kwargs)
+        except Exception:
+            _logger.debug("Failed to call chat completions", exc_info=True)
+            raise
 
-    return _call_chat_completions(user_prompt, system_prompt, use_case)
+    return _call_chat_completions(user_prompt, system_prompt, tools, model, use_case)
 
 
 def _parse_databricks_judge_response(
     llm_output: str | None,
     assessment_name: str,
+    trace: "Trace | None" = None,
 ) -> Feedback:
     """
     Parse the response from Databricks judge into a Feedback object.
 
     Args:
         llm_output: Raw output from the LLM, or None if no response.
         assessment_name: Name of the assessment.
+        trace: Optional trace object to associate with the feedback.
 
     Returns:
         Feedback object with parsed results or error.
     """
     source = AssessmentSource(
         source_type=AssessmentSourceType.LLM_JUDGE, source_id=_DATABRICKS_DEFAULT_JUDGE_MODEL
     )
+    trace_id = trace.info.trace_id if trace else None
+
     if not llm_output:
         return Feedback(
             name=assessment_name,
             error="Empty response from Databricks judge",
             source=source,
+            trace_id=trace_id,
         )
+
     try:
         response_data = json.loads(llm_output)
     except json.JSONDecodeError as e:
+        _logger.debug(f"Invalid JSON response from Databricks judge: {e}", exc_info=True)
         return Feedback(
             name=assessment_name,
-            error=f"Invalid JSON response from Databricks judge: {e}",
+            error=f"Invalid JSON response from Databricks judge: {e}\n\nLLM output: {llm_output}",
             source=source,
+            trace_id=trace_id,
         )
+
     if "result" not in response_data:
         return Feedback(
             name=assessment_name,
             error=f"Response missing 'result' field: {response_data}",
             source=source,
+            trace_id=trace_id,
         )
+
     return Feedback(
         name=assessment_name,
         value=response_data["result"],
         rationale=response_data.get("rationale", ""),
         source=source,
+        trace_id=trace_id,
+    )
+
+
+def _create_litellm_message_from_databricks_response(
```

Comment:
note that this function, `_serialize_messages_to_databricks_prompts`, and `_invoke_databricks_default_judge` are copied verbatim from the ones in `databricks_adapter.py`